micrometer-observation/src/main/java/io/micrometer/observation/SimpleObservation.java

```diff
@@ -55,6 +55,32 @@ class SimpleObservation implements Observation {
                 .filter(handler -> handler.supportsContext(this.context))
                 .collect(Collectors.toCollection(ArrayDeque::new));
         this.filters = registry.observationConfig().getObservationFilters();
+        List<ObservationConvention<?>> observationConventions = registry.observationConfig().getObservationConventions()
+                .stream().filter(observationConvention -> observationConvention.supportsContext(this.context))
+                .collect(Collectors.toList());
+        if (!observationConventions.isEmpty()) {
+            this.keyValuesProviders.addAll(observationConventions);
+            this.context.setName(observationConventions.get(0).getName());
```

Comment:
Can we do an `isBlank` check here? Let's say you don't want to provide a name so you return `null` or `""`.